### ISSUE TITLE
objc_macos_dbg_native failed to build on Mac

### ISSUE THREAD

**Issue by adelez** [grpc/grpc#13142]
/Volumes/BuildData/tmpfs/src/github/grpc/workspace_objc_macos_dbg_native/src/core/ext/transport/chttp2/transport/flow_control.cc:168:44: error: implicit conversion loses integer precision: 'int64_t' (aka 'long long') to 'const uint32_t' (aka 'const unsigned int') [-Werror,-Wshorten-64-to-32]
  const uint32_t target_announced_window = target_window();
                 ~~~~~~~~~~~~~~~~~~~~~~~   ^~~~~~~~~~~~~~~

https://sponge.corp.google.com/target?id=ac54fede-3a47-4683-9550-ecac3f447313&target=github/grpc/objc_macos_dbg_native&searchFor=&show=FAILED&sortBy=STATUS

**Comment by adelez** [comment]
The other built failure in the same run:
PhaseScriptExecution [CP]\ Check\ Pods\ Manifest.lock /Users/kbuilder/Library/Developer/Xcode/DerivedData/Sample-grxnrfbqheccwbceveswiaozemit/Build/Intermediates/Sample.build/Debug-iphonesimulator/Sample.build/Script-41F7486D8F66994B0BFB84AF.sh
    /bin/sh -c /Users/kbuilder/Library/Developer/Xcode/DerivedData/Sample-grxnrfbqheccwbceveswiaozemit/Build/Intermediates/Sample.build/Debug-iphonesimulator/Sample.build/Script-41F7486D8F66994B0BFB84AF.sh
Undefined symbols for architecture x86_64:
  "___cxa_begin_catch", referenced from:
      ___clang_call_terminate in libgRPC-Core.a(chttp2_transport.o)
  "std::terminate()", referenced from:
      ___clang_call_terminate in libgRPC-Core.a(chttp2_transport.o)
  "___gxx_personality_v0", referenced from:
      grpc_core::chttp2::StreamFlowControl::~StreamFlowControl() in libgRPC-Core.a(chttp2_transport.o)
      grpc_core::chttp2::TransportFlowControl::TransportFlowControl(grpc_exec_ctx*, grpc_chttp2_transport const*, bool) in libgRPC-Core.a(flow_control.o)
      grpc_core::chttp2::TransportFlowControl::MaybeSendUpdate(bool) in libgRPC-Core.a(flow_control.o)
      grpc_core::chttp2::StreamFlowControl::RecvData(long long) in libgRPC-Core.a(flow_control.o)
      grpc_core::chttp2::StreamFlowControl::MaybeSendUpdate() in libgRPC-Core.a(flow_control.o)
      grpc_core::chttp2::StreamFlowControl::IncomingByteStreamUpdate(unsigned long, unsigned long) in libgRPC-Core.a(flow_control.o)
      grpc_core::chttp2::FlowControlTrace::~FlowControlTrace() in libgRPC-Core.a(flow_control.o)
      ...
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

**Comment by muxi** [comment]
Fixed by #13094
